src/rt/util/container/treap.d

```diff
@@ -27,9 +27,10 @@ nothrow:
         removeAll();
     }
 
-    void initialize()
+    void initialize(ulong randSeed)
     {
-        rand48.defaultSeed();
+        Rand48 _rand48 = { randSeed };
+        rand48 = _rand48;
```

Comment:
Doesn't `rand48 = Rand48(randSeed);` work here?

Reply:
no it doesn't because of the `opCall` override

Reply:
ok (though it seems a hole in `@safe` if you are allowed to initialize any private members of a struct like this.) I'm not sure why the compiler tries to invoke a non-static opCall. Adding `this(ulong)` to `Rand48` allows to use the constructor syntax.